pkg/bootstrap/platform/platform.go

```diff
@@ -30,6 +34,13 @@ type Environment interface {
 	// Locality returns the run location for the bootstrap transformed from the
 	// platform-specific representation into the Envoy Locality schema.
 	Locality() *core.Locality
+
+	// Labels returns a collection of labels that exist on the underlying
+	// instance, structured as a map for label name to values.
+	Labels() map[string]string
+
+	// IsKubernetes determines if running on Kubernetes
+	IsKubernetes() bool
```

Comment:
Is this accurate ? The implementation seems to be GCP, not all k8s.

Second: labels are a generic concept, not specific to K8S - why do we care ? GCP or AWS VMs may also have labels, we generally avoid k8s-only behavior.

My suggestion is to drop this from the interface - and document that Labels will return platform-specific info.